scio-spanner/src/it/scala/com/spotify/scio/spanner/SpannerIOIT.scala

```diff
@@ -0,0 +1,148 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.spanner
+
+import com.google.cloud.spanner._
+import com.spotify.scio.ScioContext
+import com.spotify.scio.spanner.client.Spanner
+import org.apache.beam.sdk.io.gcp.spanner.SpannerConfig
+import org.apache.beam.sdk.options.PipelineOptionsFactory
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Await, ExecutionContext}
+import scala.concurrent.duration.Duration
+import scala.util.Random
+
+object SpannerIOIT {
+  private val projectId = "data-integration-test"
+  private val options = PipelineOptionsFactory.create()
+  private val config: SpannerConfig = SpannerConfig.create()
+    .withProjectId(projectId)
+    .withDatabaseId(s"io_it_${Random.nextInt}")
+    .withInstanceId("spanner-it")
+
+  private val adminClient = Spanner.adminClient(projectId)
+  private val dbClient = Spanner.databaseClient(config)
+
+  private final case class FakeSpannerData(asMutations: Seq[Mutation], asStructs: Seq[Struct])
+  private def fakeData(tableName: String) = FakeSpannerData(
+    Seq(
+      Mutation
+        .newInsertBuilder(tableName)
+        .set("Key").to(1L).set("Value").to("foo")
+        .build(),
+      Mutation
+        .newInsertBuilder(tableName)
+        .set("Key").to(2L).set("Value").to("bar")
+        .build()
+    ),
+    Seq(
+      Struct.newBuilder()
+        .set("Key").to(1L).set("Value").to("foo")
+        .build(),
+      Struct.newBuilder()
+        .set("Key").to(2L).set("Value").to("bar")
+        .build())
+  )
+}
+
+class SpannerIOIT extends FlatSpec with Matchers with BeforeAndAfterAll {
+  import SpannerIOIT._
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  override def beforeAll(): Unit = {
+    adminClient
+      .createDatabase(
+        config.getInstanceId.get(),
+        config.getDatabaseId.get(),
+        List(
+          "CREATE TABLE read_query_test ( Key INT64, Value STRING(MAX) ) PRIMARY KEY (Key)",
+          "CREATE TABLE read_table_test ( Key INT64, Value STRING(MAX) ) PRIMARY KEY (Key)",
+          "CREATE TABLE write_test ( Key INT64, Value STRING(MAX) ) PRIMARY KEY (Key)"
+        ).asJava)
+      .waitFor()
+  }
+
+  override def afterAll(): Unit = {
+    adminClient.dropDatabase(config.getInstanceId.get(), config.getDatabaseId.get())
+  }
+
+  private class PopulatedSpannerTable(val tableName: String) {
+    val spannerRows: FakeSpannerData = fakeData(tableName)
+    dbClient.write(spannerRows.asMutations.asJava)
+  }
+
+  private class ReadableSpannerTable(val tableName: String) {
+    val writeData: FakeSpannerData = fakeData(tableName)
+
+    lazy val readOperationResults: List[Struct] = {
+      val txn = dbClient
+        .readOnlyTransaction()
+        .read(tableName, KeySet.all(), Seq("Key", "Value").asJava)
+
+      List(
```

Comment:
@ClaireMcGinty will this brake if we change `FakeSpannerData`?

Reply:
updated it to iterate for the # of elements in fakeData